pkg/controller/logstorage/linseed.go

```diff
@@ -45,15 +45,15 @@ func (r *ReconcileLogStorage) createLinseed(
 	ctx context.Context,
 	certificateManager certificatemanager.CertificateManager,
 ) (reconcile.Result, bool, error) {
-	svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
-	svcDNSNames = append(svcDNSNames, dns.GetServiceDNSNames(linseed.ServiceName, render.ElasticsearchNamespace, r.clusterDomain)...)
-
+	// Create a KeyPair for Linseed to present to clients.
+	svcDNSNames := dns.GetServiceDNSNames(linseed.ServiceName, render.ElasticsearchNamespace, r.clusterDomain)
 	linseedKeyPair, err := certificateManager.GetOrCreateKeyPair(r.client, render.TigeraLinseedSecret, common.OperatorNamespace(), svcDNSNames)
```

Comment:
If I am not mistaken the linseedKeyPair is not actually rendered anywhere in this branch.
Like so (in esgateway.go for example):
```
	certificateComponent := rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
		Namespace:       render.ElasticsearchNamespace,
		ServiceAccounts: []string{esgateway.ServiceAccountName},
		KeyPairOptions: []rcertificatemanagement.KeyPairOption{
			rcertificatemanagement.NewKeyPairOption(gatewayKeyPair, true, true),
		},
		TrustedBundle: trustedBundle,
	})
```

Reply:
Doesn't this call to `GetOrCreateKeyPair` create the keypair if it doesn't exist?

Reply:
It fetches the existing copy from the operator namespace, which never exists in your case and because it does not, it creates a new instance. It does not store it in k8s. You still need this code snippet I made above to render it in the operator namespace and the target namespace where the pod picks it up.

Reply:
It looks like we have that code here right now: https://github.com/tigera/operator/blob/0146a0b1b2d3c9995812d617b00cd6b12dd8330a/pkg/controller/logstorage/linseed.go#L107-L114

Reply:
That is a problem. The trustedbundle should only be rendered once, otherwise two controllers are competing and the last one wins.

I think it should probably be rendered from the logstorage_controller and not from esgateway.go and here.